src/test/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadAndUpdateOperationTest.java

```diff
@@ -144,53 +145,708 @@ public void findAndUpdate() throws Exception {
+            Values.of(CustomValueSerializers.getBooleanMapValues(shredDocument.queryBoolValues())),
+            Values.of(CustomValueSerializers.getDoubleMapValues(shredDocument.queryNumberValues())),
+            Values.of(CustomValueSerializers.getStringMapValues(shredDocument.queryTextValues())),
+            Values.of(CustomValueSerializers.getSetValue(shredDocument.queryNullValues())),
+            Values.of(shredDocument.docJson()),
+            Values.of(CustomValueSerializers.getDocumentIdValue(shredDocument.id())),
+            Values.of(tx_id2))
+        .withColumnSpec(
+            List.of(
+                QueryOuterClass.ColumnSpec.newBuilder()
+                    .setName("applied")
+                    .setType(TypeSpecs.BOOLEAN)
+                    .build()))
+        .returning(List.of(List.of(Values.of(true))));
+
+    String updater =
+        """
+                  {
+                    "findOneAndUpdate": {
+                      "filter": {
+                        "username": "user1"
+                      },
+                      "update": {
+                        "$set": {
+                          "name": "test"
+                        }
+                      }
+                    }
+                  }
+                """;
+
+    FindOneAndUpdateCommand findOneAndUpdateCommand =
+        objectMapper.readValue(updater, FindOneAndUpdateCommand.class);
+    ReadOperation readOperation =
+        new FindOperation(
+            commandContext,
+            List.of(
+                new DBFilterBase.TextFilter(
+                    "username", DBFilterBase.MapFilterBase.Operator.EQ, "user1")),
+            null,
+            1,
+            1,
+            ReadType.DOCUMENT,
+            objectMapper);
+    DocumentUpdater documentUpdater =
+        DocumentUpdater.construct(findOneAndUpdateCommand.updateClause());
+    ReadAndUpdateOperation operation =
+        new ReadAndUpdateOperation(
+            commandContext, readOperation, documentUpdater, true, false, false, shredder, 1, 3);
+    final Uni<Supplier<CommandResult>> execute = operation.execute(queryExecutor);
+    final CommandResult commandResultSupplier = execute.subscribe().asCompletionStage().get().get();
+    UniAssertSubscriber<Supplier<CommandResult>> subscriber =
+        operation.execute(queryExecutor).subscribe().withSubscriber(UniAssertSubscriber.create());
+    assertThat(commandResultSupplier)
+        .satisfies(
+            commandResult -> {
+              assertThat(commandResultSupplier.status()).isNotNull();
+              assertThat(commandResultSupplier.status().get(CommandStatus.MATCHED_COUNT))
+                  .isEqualTo(1);
+              assertThat(commandResultSupplier.status().get(CommandStatus.MODIFIED_COUNT))
+                  .isEqualTo(1);
+              assertThat(commandResultSupplier.errors()).isNull();
+            });
+  }
+
+  @Test
+  public void findAndUpdateWithRetryFailure() throws Exception {
```

Comment:
Ok I can see verification of LWT failure, but not sure how failure itself is injected in this test?

Reply:
LWT update returns boolean in response. If the response is false it means lwt failure. It's done with below statement for update response.

.returning(List.of(List.of(Values.of(false))));

Reply:
Added comments in the test class